### PR TITLE
comunity: Implement delete method and all async methods in opensearch_vector_search

### PR DESCRIPTION
**Pull request by 2jimoo**
- **Description:** In order to use index and aindex in libs/langchain/langchain/indexes/_api.py, I implemented delete method and all async methods in opensearch_vector_search
- **Dependencies:** No changes

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
cc @naveentatikonda

**Comment by 2jimoo** [comment]
@baskaryan  May I ask if the PR is under review?

**Comment by KaganAddSearch** [comment]
Hey, I wish this PR was merged. @2jimoo @baskaryan

**Comment by 2jimoo** [comment]
Oh, I'm glad you commented like that. I'll open it again.

File: libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py [hunk 1]
```diff
@@ -81,5 +81,6 @@ def check_compatibility(vector_store: VectorStore) -> bool:
         "ZepVectorStore",
         "Zilliz",
         "Lantern",
+        "OpenSearchVectorSearch",
```
Comment (baskaryan) [inline libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:84]: can we add OpenSearchVectorStore to the list in docs/docs/modules/data_conections/indexing.ipynb

Comment (2jimoo) [reply libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:84]: I'll do that, and I have a question.
Should I change the name OpenSearchVectorSearch to OpenSearchVectorStore?

Comment (baskaryan) [reply libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:94]: yep, sorry typo on my end. Let's keep as OpenSearchVectorSearch

**Review by baskaryan — APPROVED** [review]

**Pull request merged**